1.0.0及以上版本构建 image 时缺少 rimraf 模块

`0.1.6` 版本运行正常。如需提供其他信息请回复，感谢。

Dockerfile 内容如下：
```bash
FROM node:16.18.1
WORKDIR /usr/src/app

COPY package*.json ./
RUN npm ci --omit=dev --ignore-scripts && \
  npm i -g nodeinstall && \
  nodeinstall --install-alinode ^7

COPY . .
EXPOSE 7001
# USER node
CMD npm run start
```

报错内容如下：
```bash
...
Step 4/7 : RUN npm ci --omit=dev --ignore-scripts &&   npm i -g nodeinstall@1.0.0 &&   nodeinstall --install-alinode ^7
 ---> Running in c61bd2d19df3
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.

added 592 packages, and audited 593 packages in 19s

found 0 vulnerabilities

added 104 packages, and audited 105 packages in 58s

6 packages are looking for funding
  run `npm fund` for details

6 vulnerabilities (2 low, 4 high)

To address issues that do not require attention, run:
  npm audit fix

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
npm notice 
npm notice New major version of npm available! 8.19.2 -> 9.1.3
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.1.3>
npm notice Run `npm install -g npm@9.1.3` to update!
npm notice 
node:internal/modules/cjs/loader:988
  throw err;
  ^

Error: Cannot find module 'rimraf'
Require stack:
- /usr/local/lib/node_modules/nodeinstall/lib/installer/installer.js
- /usr/local/lib/node_modules/nodeinstall/lib/installer/node_installer.js
- /usr/local/lib/node_modules/nodeinstall/lib/config.js
- /usr/local/lib/node_modules/nodeinstall/lib/install.js
- /usr/local/lib/node_modules/nodeinstall/bin/nodeinstall.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:985:15)
    at Function.Module._load (node:internal/modules/cjs/loader:833:27)
    at Module.require (node:internal/modules/cjs/loader:1057:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/nodeinstall/lib/installer/installer.js:15:16)
    at Module._compile (node:internal/modules/cjs/loader:1155:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
    at Module.load (node:internal/modules/cjs/loader:1033:32)
    at Function.Module._load (node:internal/modules/cjs/loader:868:12)
    at Module.require (node:internal/modules/cjs/loader:1057:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/usr/local/lib/node_modules/nodeinstall/lib/installer/installer.js',
    '/usr/local/lib/node_modules/nodeinstall/lib/installer/node_installer.js',
    '/usr/local/lib/node_modules/nodeinstall/lib/config.js',
    '/usr/local/lib/node_modules/nodeinstall/lib/install.js',
    '/usr/local/lib/node_modules/nodeinstall/bin/nodeinstall.js'
  ]
}
The command '/bin/sh -c npm ci --omit=dev --ignore-scripts &&   npm i -g nodeinstall@1.0.0 &&   nodeinstall --install-alinode ^7' returned a non-zero code: 1
```

--- a/README.md
+++ b/README.md
@@ -83,12 +83,9 @@ Also support
 ## API
 
 ```js
-const co = require('co');
 const install = require('nodeinstall').install;
-co(function* () {
-  yield install({
-    version: '^18.0.0',
-  });
+await install({
+  version: '^18.0.0',
 });
 ```
 

--- a/bin/nodeinstall.js
+++ b/bin/nodeinstall.js
@@ -2,7 +2,6 @@
 
 'use strict';
 
-const co = require('co');
 const install = require('../lib/install');
 const program = require('commander');
 const only = require('only');
@@ -26,19 +25,19 @@ if (!(program.args[0] || program.installNightly)) {
   process.exit(1);
 }
 
-co(function* () {
-  const options = only(program, [
-    'installNode',
-    'installNoderc',
-    'installAlinode',
-    'installNsolid',
-    'installNightly',
-    'china',
-    'cache',
-    'distUrl',
-  ]);
-  options.version = program.args[0];
-  yield install(options);
+const options = only(program, [
+  'installNode',
+  'installNoderc',
+  'installAlinode',
+  'installNsolid',
+  'installNightly',
+  'china',
+  'cache',
+  'distUrl',
+]);
+options.version = program.args[0];
+install(options).then(() => {
+  process.exit(0);
 }).catch(err => {
   console.error(err.stack);
   process.exit(1);

--- a/lib/install.js
+++ b/lib/install.js
@@ -20,7 +20,7 @@ const DEFAULT_OPTIONS = {
 };
 const TYPES = Object.keys(config);
 
-module.exports = function* (options) {
+module.exports = async function install(options) {
   options = extend({}, DEFAULT_OPTIONS, options);
   const isChina = options.china || process.env.NODEINSTALL_CHINA;
   const result = getTypeAndVersion(options);
@@ -34,7 +34,7 @@ module.exports = function* (options) {
 
   if (type === 'nightly') {
     // always get the latest version
-    version = yield getNightlyVersion();
+    version = await getNightlyVersion();
   }
 
   const installer = new Installer({
@@ -42,7 +42,7 @@ module.exports = function* (options) {
     distUrl,
     version,
   });
-  yield installer.install();
+  await installer.install();
 };
 
 function getTypeAndVersion(options) {

--- a/lib/installer/alinode_installer.js
+++ b/lib/installer/alinode_installer.js
@@ -10,8 +10,8 @@ class AlinodeInstaller extends Installer {
     super(Object.assign({}, options, { name: 'alinode' }));
   }
 
-  * getVersion() {
-    return yield getAlinodeVersion(this.options);
+  async getVersion() {
+    return await getAlinodeVersion(this.options);
   }
 }
 

--- a/lib/installer/installer.js
+++ b/lib/installer/installer.js
@@ -222,7 +222,7 @@ async function putBinaryWin32(nodeDir, projectBinDir) {
     const source = path.resolve(nodeDir, file);
     const target = path.resolve(projectBinDir, file);
     if (fs.existsSync(target)) {
-      fs.promises.rm(target, { recursive: true, force: true });
+      await fs.promises.rm(target, { recursive: true, force: true });
     }
 
     if (!fs.statSync(source).isDirectory()) {

--- a/lib/installer/installer.js
+++ b/lib/installer/installer.js
@@ -10,9 +10,6 @@ const crypto = require('crypto');
 const zlib = require('zlib');
 const tar = require('tar');
 const child_process = require('child_process');
-
-const mkdirp = require('mkdirp');
-const rimraf = require('rimraf');
 const debug = require('debug')('nodeinstall');
 const assert = require('assert');
 const AdmZip = require('adm-zip');
@@ -34,7 +31,7 @@ class Installer {
     debug('Start install %s@%s, options: %j', options.name, options.version, options);
   }
 
-  * install() {
+  async install() {
     const cwd = this.options.cwd;
     const name = this.options.name;
     const distUrl = this.options.distUrl;
@@ -45,7 +42,7 @@ class Installer {
     const nodeLink = path.join(nodeLinkDir, 'node');
     const npmLink = path.join(nodeLinkDir, 'npm');
 
-    let version = yield this.getVersion();
+    let version = await this.getVersion();
 
     try {
       version = getSafeVersion(version, this.options.unsafeVersions);
@@ -56,7 +53,7 @@ class Installer {
 
     try {
       // process.versions from installed node
-      const versions = getLocalNodeVersion(cwd);
+      const versions = await getLocalNodeVersion(cwd);
       if (version === versions[name]) {
         console.info('%s has been installed, version %s', name, version);
         return;
@@ -67,6 +64,7 @@ class Installer {
       }
     }
 
+    await fs.promises.mkdir(nodeDir, { recursive: true });
     let platform = process.env.MOCK_OS_PLATFORM || os.platform();
     if (platform === 'win32') platform = 'win';
     const arch = os.arch() === 'ia32' ? 'x86' : os.arch();
@@ -83,25 +81,25 @@ class Installer {
       // download tarball if cache is disabled or cache file is not exist,
       // otherwise use cache
       if (cacheStrategy.disable || !fs.existsSync(targetFile)) {
-        yield downloadFile(pkgUrl, shaUrl, targetFile);
+        await downloadFile(pkgUrl, shaUrl, targetFile);
       }
 
       // extract tarball/zip to $cwd/node_modules
       if (platform === 'win') {
         const entryDir = path.basename(targetFile, '.zip');
         const destDir = path.resolve(nodeLinkDir, entryDir);
-        rimraf.sync(destDir);
+        await fs.promises.rm(destDir, { recursive: true, force: true });
 
         try {
-          yield unzipFile(targetFile, path.dirname(destDir));
+          await unzipFile(targetFile, path.dirname(destDir));
         } catch (e) {
           console.error('[install] extract zip error:', e);
         }
 
-        rimraf.sync(nodeDir);
+        await fs.promises.rm(nodeDir, { recursive: true, force: true });
         cpWin32(destDir, nodeDir, false);
       } else {
-        yield extract(targetFile, nodeDir);
+        await extract(targetFile, nodeDir);
       }
     } catch (err) {
       if (fs.existsSync(targetFile)) {
@@ -110,9 +108,9 @@ class Installer {
       throw err;
     }
 
-    mkdirp.sync(nodeLinkDir);
+    await fs.promises.mkdir(nodeLinkDir, { recursive: true });
     if (platform === 'win') {
-      putBinaryWin32(nodeDir, nodeLinkDir);
+      await putBinaryWin32(nodeDir, nodeLinkDir);
     } else {
       if (!fs.existsSync(nodeLink)) {
         fs.symlinkSync(path.normalize('../node/bin/node'), nodeLink);
@@ -128,16 +126,16 @@ class Installer {
     }, null, 2));
   }
 
-  * getVersion() {
+  async getVersion() {
     return this.options.version;
   }
 }
 
 module.exports = Installer;
 
-function* downloadFile(url, shaUrl, targetFile) {
+async function downloadFile(url, shaUrl, targetFile) {
   debug('download %s to %s', url, targetFile);
-  const ret = yield request(url, {
+  const ret = await request(url, {
     timeout: 20000,
     streaming: true,
     followRedirect: true,
@@ -152,12 +150,12 @@ function* downloadFile(url, shaUrl, targetFile) {
     throw err;
   }
 
-  yield writeFile(res, targetFile);
-  yield verifyChecksum(shaUrl, targetFile);
+  await writeFile(res, targetFile);
+  await verifyChecksum(shaUrl, targetFile);
 }
 
-function* verifyChecksum(shaUrl, targetFile) {
-  const ret = yield request(shaUrl, {
+async function verifyChecksum(shaUrl, targetFile) {
+  const ret = await request(shaUrl, {
     followRedirect: true,
   });
   const res = ret.res;
@@ -213,38 +211,41 @@ function cpWin32(source, target, keepbase = false) {
   }
 }
 
-function putBinaryWin32(nodeDir, projectBinDir) {
-  [
+async function putBinaryWin32(nodeDir, projectBinDir) {
+  const files = [
     'node.exe',
     'npm.cmd',
     'npm',
     'node_modules',
-  ].forEach(function(file) {
+  ];
+  for (const file of files) {
     const source = path.resolve(nodeDir, file);
     const target = path.resolve(projectBinDir, file);
-    if (fs.existsSync(target)) rimraf.sync(target);
+    if (fs.existsSync(target)) {
+      fs.promises.rm(target, { recursive: true, force: true });
+    }
 
     if (!fs.statSync(source).isDirectory()) {
       cpWin32(source, projectBinDir, true);
     } else {
       cpWin32(source, target, false);
     }
-  });
+  }
 }
 
 function extract(targetFile, nodeDir) {
   return new Promise((resolve, reject) => {
     const gunzip = zlib.createGunzip();
     gunzip.on('error', reject);
-    const extracter = tar.Extract({ path: nodeDir, strip: 1 });
+    const extracter = tar.x({ cwd: nodeDir, strip: 1 });
     extracter.on('error', reject);
     extracter.on('end', resolve);
     fs.createReadStream(targetFile).pipe(gunzip).pipe(extracter);
   });
 }
 
-function writeFile(res, targetFile) {
-  mkdirp.sync(path.dirname(targetFile));
+async function writeFile(res, targetFile) {
+  await fs.promises.mkdir(path.dirname(targetFile), { recursive: true });
   return new Promise(function(resolve, reject) {
     const ws = fs.createWriteStream(targetFile)
       .on('error', reject)

--- a/lib/installer/node_installer.js
+++ b/lib/installer/node_installer.js
@@ -10,8 +10,8 @@ class NodeInstaller extends Installer {
     super(options);
   }
 
-  * getVersion() {
-    return yield getNodeVersion(this.options);
+  async getVersion() {
+    return await getNodeVersion(this.options);
   }
 }
 

--- a/lib/installer/nsolid_installer.js
+++ b/lib/installer/nsolid_installer.js
@@ -11,12 +11,12 @@ class NSolidInstaller extends Installer {
     super(Object.assign({}, options, { name: 'nsolid' }));
   }
 
-  * getVersion() {
-    return yield getNodeVersion(this.options);
+  async getVersion() {
+    return await getNodeVersion(this.options);
   }
 
-  * install() {
-    yield super.install();
+  async install() {
+    await super.install();
     fs.symlinkSync(path.normalize('./node'), path.join(this.options.cwd, 'node_modules/.bin/nsolid'));
   }
 }

--- a/lib/request.js
+++ b/lib/request.js
@@ -2,19 +2,19 @@
 
 const urllib = require('urllib');
 
-module.exports = function* request(url, options) {
+module.exports = async function request(url, options) {
   options = options || {};
   const retry = Number(options.retry) || 0;
-  return yield _request(url, options, retry);
+  return await _request(url, options, retry);
 };
 
-function* _request(url, options, retry) {
+async function _request(url, options, retry) {
   try {
-    return yield urllib.request(url, options);
+    return await urllib.request(url, options);
   } catch (e) {
     if (retry > 0) {
       retry--;
-      return yield _request(url, options, retry);
+      return await _request(url, options, retry);
     }
     throw e;
   }

--- a/lib/version.js
+++ b/lib/version.js
@@ -13,23 +13,23 @@ const request = require('./request');
 
 module.exports = {
 
-  * getAlinodeVersion(options) {
+  async getAlinodeVersion(options) {
     options = options || {};
     const distUrl = options.distUrl;
     assert(distUrl, 'distUrl is required');
-    const versions = yield getAlinodeVersions(distUrl);
+    const versions = await getAlinodeVersions(distUrl);
     return _getNodeVersion(options.version, versions);
   },
 
-  * getNodeVersion(options) {
+  async getNodeVersion(options) {
     options = options || {};
     const distUrl = options.distUrl;
     assert(distUrl, 'distUrl is required');
-    const versions = yield getNodeVersions(distUrl);
+    const versions = await getNodeVersions(distUrl);
     return _getNodeVersion(options.version, versions);
   },
 
-  * getLocalNodeVersion(cwd) {
+  async getLocalNodeVersion(cwd) {
     const nodeBinDir = path.join(cwd, 'node_modules', 'node', 'bin');
     let nodebin;
     if (os.platform() === 'win32') {
@@ -48,8 +48,8 @@ module.exports = {
     return JSON.parse(stdout);
   },
 
-  * getNightlyVersion() {
-    return yield nodeNightlyVersion();
+  async getNightlyVersion() {
+    return await nodeNightlyVersion();
   },
 
   getSafeVersion(version, unsafeVersions) {
@@ -89,8 +89,8 @@ function _getNodeVersion(version, versions) {
   throw new Error(`${version} is not matched`);
 }
 
-function* getNodeVersions(distUrl) {
-  const res = yield request(distUrl + '/index.json', {
+async function getNodeVersions(distUrl) {
+  const res = await request(distUrl + '/index.json', {
     method: 'GET',
     followRedirect: true,
     dataType: 'json',
@@ -103,8 +103,8 @@ function* getNodeVersions(distUrl) {
   return versions;
 }
 
-function* getAlinodeVersions(distUrl) {
-  const res = yield request(distUrl + '/index.tab', {
+async function getAlinodeVersions(distUrl) {
+  const res = await request(distUrl + '/index.tab', {
     timeout: 10000,
     method: 'GET',
     followRedirect: true,

--- a/package.json
+++ b/package.json
@@ -5,11 +5,9 @@
   "dependencies": {
     "adm-zip": "^0.5.9",
     "bytes": "^2.5.0",
-    "co": "^4.6.0",
     "commander": "^2.9.0",
     "debug": "^2.6.6",
     "extend": "^3.0.1",
-    "mkdirp": "^0.5.1",
     "node-nightly-version": "^1.0.6",
     "only": "^0.0.2",
     "progress": "^2.0.0",
@@ -23,8 +21,7 @@
     "egg-bin": "^5.5.0",
     "eslint": "^8.28.0",
     "eslint-config-egg": "^12.1.0",
-    "git-contributor": "^1.0.10",
-    "rimraf": "^2.6.1"
+    "git-contributor": "^1.0.10"
   },
   "main": "lib/index.js",
   "files": [

--- a/test-win32/install_node_win32.test.js
+++ b/test-win32/install_node_win32.test.js
@@ -3,7 +3,7 @@
 const fs = require('fs');
 const path = require('path');
 const assert = require('assert');
-const rimraf = require('rimraf');
+const fsPromises = require('fs/promises');
 const coffee = require('coffee');
 const execSync = require('child_process').execSync;
 
@@ -25,21 +25,21 @@ function getEnv(nodeBinPath) {
 
 describe('test/install_node_win32.test.js', function() {
   let cwd = path.join(fixtures, 'install-node');
-  beforeEach(function() {
+  beforeEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
-  afterEach(function() {
+  afterEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
 
-  it('should install node', function* () {
+  it('should install node', async function() {
     cwd = path.join(fixtures, 'install-node');
     // zip file for windows provided started from nodejs 6.2.2
-    yield coffee
+    await coffee
       .fork(nodeinstall, [ '6.2.2' ], { cwd })
       .debug()
       .expect('code', 0)
@@ -57,9 +57,9 @@ describe('test/install_node_win32.test.js', function() {
     assert(_trimEOL(execSync(`\"${npmBinPath}\" -v`, { env: getEnv(nodeBinPath) }).toString()) === '3.9.5');
   });
 
-  it('should install noderc', function* () {
+  it('should install noderc', async function() {
     cwd = path.join(fixtures, 'install-node');
-    yield coffee
+    await coffee
       .fork(nodeinstall, [ '--install-noderc', '6.9.2-rc.1' ], { cwd })
       .debug()
       .expect('code', 0)
@@ -75,9 +75,9 @@ describe('test/install_node_win32.test.js', function() {
     assert(_trimEOL(execSync(`${nodeBinPath} -v`).toString()) === 'v6.9.2-rc.1');
   });
 
-  it('should install node nightly', function* () {
+  it('should install node nightly', async function() {
     cwd = path.join(fixtures, 'install-node');
-    yield coffee
+    await coffee
       .fork(nodeinstall, [ '--install-nightly' ], { cwd })
       .debug()
       .expect('code', 0)

--- a/test/install_alinode.test.js
+++ b/test/install_alinode.test.js
@@ -3,7 +3,7 @@
 const fs = require('fs');
 const path = require('path');
 const assert = require('assert');
-const rimraf = require('rimraf');
+const fsPromises = require('fs/promises');
 const execSync = require('child_process').execSync;
 const coffee = require('coffee');
 const nodeinstall = path.join(__dirname, '../bin/nodeinstall');
@@ -13,23 +13,23 @@ const fixtures = path.join(__dirname, 'fixtures');
 
 describe.skip('test/install_alinode.test.js', function() {
   let cwd;
-  beforeEach(function() {
+  beforeEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
-  afterEach(function() {
+  afterEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
 
-  it('should install alinode', function* () {
+  it('should install alinode', async function() {
     this.timeout(1000000);
 
     cwd = path.join(fixtures, 'install-alinode');
 
-    yield coffee
+    await coffee
       .fork(nodeinstall, [ '--install-alinode', '1.4.0' ], { cwd })
       .debug()
       .expect('code', 0)

--- a/test/install_node.test.js
+++ b/test/install_node.test.js
@@ -3,7 +3,7 @@
 const fs = require('fs');
 const path = require('path');
 const assert = require('assert');
-const rimraf = require('rimraf');
+const fsPromises = require('fs/promises');
 const coffee = require('coffee');
 const execSync = require('child_process').execSync;
 
@@ -13,20 +13,20 @@ const nodeinstall = path.join(__dirname, '../bin/nodeinstall');
 
 describe('test/install_node.test.js', function() {
   let cwd;
-  beforeEach(function() {
+  beforeEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
-  afterEach(function() {
+  afterEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      // await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
 
-  it('should install node', function* () {
+  it('should install node', async function() {
     cwd = path.join(fixtures, 'install-node');
-    yield coffee
+    await coffee
       .fork(nodeinstall, [ '4.0.0' ], { cwd })
       .debug()
       .expect('code', 0)
@@ -45,9 +45,9 @@ describe('test/install_node.test.js', function() {
     assert(execSync(`${npmBinPath} -v`).toString() === '2.14.2\n');
   });
 
-  it('should install noderc', function* () {
+  it('should install noderc', async function() {
     cwd = path.join(fixtures, 'install-node');
-    yield coffee
+    await coffee
       .fork(nodeinstall, [ '--install-noderc', '6.0.0-rc.3' ], { cwd })
       .debug()
       .expect('code', 0)
@@ -66,9 +66,9 @@ describe('test/install_node.test.js', function() {
   });
 
 
-  it('should install node nightly', function* () {
+  it('should install node nightly', async function() {
     cwd = path.join(fixtures, 'install-node');
-    yield coffee
+    await coffee
       .fork(nodeinstall, [ '--install-nightly' ], { cwd })
       .debug()
       .expect('code', 0)

--- a/test/install_nsolid.test.js
+++ b/test/install_nsolid.test.js
@@ -3,7 +3,7 @@
 const fs = require('fs');
 const path = require('path');
 const assert = require('assert');
-const rimraf = require('rimraf');
+const fsPromises = require('fs/promises');
 const execSync = require('child_process').execSync;
 const coffee = require('coffee');
 
@@ -13,21 +13,21 @@ const nodeinstall = path.join(__dirname, '../bin/nodeinstall');
 
 describe('test/install_nsolid.test.js', function() {
   let cwd;
-  beforeEach(function() {
+  beforeEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
-  afterEach(function() {
+  afterEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
 
-  it('should install-nsolid', function* () {
+  it('should install-nsolid', async function() {
     cwd = path.join(fixtures, 'install-nsolid');
 
-    yield coffee
+    await coffee
       .fork(nodeinstall, [ '--install-nsolid', '1.3.2' ], { cwd })
       .debug()
       .expect('code', 0)

--- a/test/installer.test.js
+++ b/test/installer.test.js
@@ -1,7 +1,7 @@
 'use strict';
 
 const path = require('path');
-const rimraf = require('rimraf');
+const fsPromises = require('fs/promises');
 const assert = require('assert');
 const execSync = require('child_process').execSync;
 const NodeInstaller = require('..').NodeInstaller;
@@ -11,20 +11,20 @@ const getDistUrl = require('./utils').getDistUrl;
 
 describe('test/installer.test.js', function() {
   let cwd;
-  beforeEach(function() {
+  beforeEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
-  afterEach(function() {
+  afterEach(async function() {
     if (cwd) {
-      rimraf.sync(path.join(cwd, 'node_modules'));
+      await fsPromises.rm(path.join(cwd, 'node_modules'), { recursive: true, force: true });
     }
   });
 
   describe('unsafeVersions', function() {
 
-    it('should match unsafeVersions', function* () {
+    it('should match unsafeVersions', async function() {
       cwd = path.join(fixtures, 'install-node');
       const nodeBinPath = path.join(cwd, 'node_modules/.bin/node');
       const installer = new NodeInstaller({
@@ -35,7 +35,7 @@ describe('test/installer.test.js', function() {
           '>= 1.0.0 < 4.4.4': '4.5.0',
         },
       });
-      yield installer.install();
+      await installer.install();
       assert(execSync(`${nodeBinPath} -v`).toString() === 'v4.5.0\n');
     });
   });

--- a/test/version.test.js
+++ b/test/version.test.js
@@ -10,23 +10,23 @@ describe('test/version.test.js', () => {
   describe('getNodeVersion', () => {
     const distUrl = getDistUrl('node');
 
-    it('should get the latest version', function* () {
-      const v = yield version.getNodeVersion({
+    it('should get the latest version', async function() {
+      const v = await version.getNodeVersion({
         distUrl,
       });
       assert(v, '');
     });
 
-    it('should get the real version that match semver', function* () {
-      const v = yield version.getNodeVersion({
+    it('should get the real version that match semver', async function() {
+      const v = await version.getNodeVersion({
         version: '~4.1.0',
         distUrl,
       });
       assert(v === '4.1.2');
     });
 
-    it('should get the real version', function* () {
-      const v = yield version.getNodeVersion({
+    it('should get the real version', async function() {
+      const v = await version.getNodeVersion({
         version: '4.1.0',
         distUrl,
       });
@@ -37,26 +37,26 @@ describe('test/version.test.js', () => {
   describe('getAlinodeVersion', () => {
     const distUrl = getDistUrl('alinode');
 
-    it('should get the latest version', function* () {
+    it('should get the latest version', async function() {
       this.timeout(1000000);
-      const v = yield version.getAlinodeVersion({
+      const v = await version.getAlinodeVersion({
         distUrl,
       });
       assert(v, '');
     });
 
-    it('should get the real version that match semver', function* () {
+    it('should get the real version that match semver', async function() {
       this.timeout(1000000);
-      const v = yield version.getAlinodeVersion({
+      const v = await version.getAlinodeVersion({
         version: '~1.2.0',
         distUrl,
       });
       assert(v === '1.2.3');
     });
 
-    it('should get the real version', function* () {
+    it('should get the real version', async function() {
       this.timeout(1000000);
-      const v = yield version.getAlinodeVersion({
+      const v = await version.getAlinodeVersion({
         version: '1.5.6',
         distUrl,
       });

